Workflow failure: Admin UI E2E - realm_test.spec.ts

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

[Admin UI E2E (4, firefox)](https://github.com/keycloak/keycloak/actions/runs/7619085643/job/20751706303)

```
1) Realm tests
       Should delete Test Disabled realm:
     AssertionError: Timed out retrying after 30000ms: Expected to find content: 'Test-Disabled' within the element: <div#realm-select.pf-c-dropdown.pf-m-expanded.keycloak__realm_selector__dropdown> but never did.
      at goToRealm (webpack://admin-ui/./cypress/support/pages/admin-ui/SidebarPage.ts:38:0)
      at  (webpack://admin-ui/./cypress/e2e/realm_test.spec.ts:59:0)
      at __webpack_modules__</setRunnable/runnable.fn (http://localhost:8080/__cypress/runner/cypress_runner.js:138559:43)
      at callFn (http://localhost:8080/__cypress/runner/cypress_runner.js:157180:21)
      at __webpack_modules__</Runnable.prototype.run (http://localhost:8080/__cypress/runner/cypress_runner.js:157167:13)
      at __webpack_modules__</create/onRunnableRun (http://localhost:8080/__cypress/runner/cypress_runner.js:163091:30)
      at finallyHandler (http://localhost:8080/__cypress/runner/cypress_runner.js:4072:23)
      at tryCatcher (http://localhost:8080/__cypress/runner/cypress_runner.js:1807:23)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromiseFromHandler (http://localhost:8080/__cypress/runner/cypress_runner.js:1519:31)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromise (http://localhost:8080/__cypress/runner/cypress_runner.js:1576:18)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromise0 (http://localhost:8080/__cypress/runner/cypress_runner.js:1621:10)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromises (http://localhost:8080/__cypress/runner/cypress_runner.js:1701:18)
      at _drainQueueStep (http://localhost:8080/__cypress/runner/cypress_runner.js:2407:12)
      at _drainQueue (http://localhost:8080/__cypress/runner/cypress_runner.js:2400:24)
      at __webpack_modules__</Async.prototype._drainQueues (http://localhost:8080/__cypress/runner/cypress_runner.js:2416:16)
      at __webpack_modules__</Async/this.drainQueues (http://localhost:8080/__cypress/runner/cypress_runner.js:2286:14)

  2) Realm tests
       should create realm from new a realm:
     AssertionError: Timed out retrying after 30000ms: Expected to find element: `.pf-c-alert__title`, but never found it. Queried from:

              > cy.findByTestId(global-alerts)
      at checkNotificationMessage (webpack://admin-ui/./cypress/support/pages/admin-ui/Masthead.ts:89:0)
      at  (webpack://admin-ui/./cypress/e2e/realm_test.spec.ts:70:0)
      at __webpack_modules__</setRunnable/runnable.fn (http://localhost:8080/__cypress/runner/cypress_runner.js:138559:43)
      at callFn (http://localhost:8080/__cypress/runner/cypress_runner.js:157180:21)
      at __webpack_modules__</Runnable.prototype.run (http://localhost:8080/__cypress/runner/cypress_runner.js:157167:13)
      at __webpack_modules__</create/onRunnableRun (http://localhost:8080/__cypress/runner/cypress_runner.js:163091:30)
      at finallyHandler (http://localhost:8080/__cypress/runner/cypress_runner.js:4072:23)
      at tryCatcher (http://localhost:8080/__cypress/runner/cypress_runner.js:1807:23)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromiseFromHandler (http://localhost:8080/__cypress/runner/cypress_runner.js:1519:31)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromise (http://localhost:8080/__cypress/runner/cypress_runner.js:1576:18)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromise0 (http://localhost:8080/__cypress/runner/cypress_runner.js:1621:10)
      at __webpack_modules__</module.exports/Promise.prototype._settlePromises (http://localhost:8080/__cypress/runner/cypress_runner.js:1701:18)
      at _drainQueueStep (http://localhost:8080/__cypress/runner/cypress_runner.js:2407:12)
      at _drainQueue (http://localhost:8080/__cypress/runner/cypress_runner.js:2400:24)
      at __webpack_modules__</Async.prototype._drainQueues (http://localhost:8080/__cypress/runner/cypress_runner.js:2416:16)
      at __webpack_modules__</Async/this.drainQueues (http://localhost:8080/__cypress/runner/cypress_runner.js:2286:14)
```

### Version

latest

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/js/apps/admin-ui/cypress/e2e/realm_test.spec.ts
+++ b/js/apps/admin-ui/cypress/e2e/realm_test.spec.ts
@@ -34,21 +34,18 @@ describe("Realm tests", () => {
     ),
   );
 
-  it("should fail creating Master realm", () => {
+  it("should fail creating duplicated or empty name realm", () => {
     sidebarPage.goToCreateRealm();
+
+    createRealmPage.createRealm().verifyRealmNameFieldInvalid();
+
     createRealmPage.fillRealmName("master").createRealm();
 
     masthead.checkNotificationMessage(
       "Could not create realm Conflict detected. See logs for details",
     );
+
     createRealmPage.cancelRealmCreation();
-  });
-
-  it("should fail creating realm with empty name", () => {
-    sidebarPage.goToCreateRealm();
-    createRealmPage.createRealm();
-
-    createRealmPage.verifyRealmNameFieldInvalid();
   });
 
   it("should create Test realm", () => {
@@ -63,24 +60,27 @@ describe("Realm tests", () => {
     masthead.checkNotificationMessage("Realm created successfully");
   });
 
-  it("should create Test Disabled realm", () => {
+  it("CRUD test of Disabled realm", () => {
     sidebarPage.goToCreateRealm();
     sidebarPage.waitForPageLoad();
 
     createRealmPage.fillRealmName(testDisabledName).createRealm();
-    createRealmPage.disableRealm();
 
     masthead.checkNotificationMessage("Realm created successfully");
-  });
 
-  it("Should cancel deleting Test Disabled realm", () => {
+    cy.reload();
+    sidebarPage.goToRealm(testDisabledName).goToRealmSettings();
+
+    createRealmPage.disableRealm();
+    modalUtils.confirmModal();
+
+    masthead.checkNotificationMessage("Realm successfully updated");
+
     sidebarPage.goToRealm(testDisabledName).goToRealmSettings();
     realmSettings.clickActionMenu();
     cy.findByText("Delete").click();
     modalUtils.cancelModal();
-  });
 
-  it("Should delete Test Disabled realm", () => {
     cy.reload();
     sidebarPage.waitForPageLoad();
     sidebarPage.goToRealm(testDisabledName).goToRealmSettings();
@@ -90,7 +90,7 @@ describe("Realm tests", () => {
     masthead.checkNotificationMessage("The realm has been deleted");
 
     // Show current realms
-    sidebarPage.showCurrentRealms(2);
+    sidebarPage.realmExists("Test-Disabled", false);
   });
 
   it("should create realm from new a realm", () => {

--- a/js/apps/admin-ui/cypress/support/pages/admin-ui/CreateRealmPage.ts
+++ b/js/apps/admin-ui/cypress/support/pages/admin-ui/CreateRealmPage.ts
@@ -2,7 +2,7 @@ export default class CreateRealmPage {
   #clearBtn = ".pf-c-file-upload__file-select button:last-child";
   #modalClearBtn = "clear-button";
   #realmNameInput = "#kc-realm-name";
-  #enabledSwitch = '[for="kc-realm-enabled-switch"] span.pf-c-switch__toggle';
+  #enabledSwitch = ".pf-c-toolbar .pf-c-switch__toggle";
   #createBtn = '.pf-c-form__group:last-child button[type="submit"]';
   #cancelBtn = '.pf-c-form__group:last-child button[type="button"]';
   #codeEditor = ".pf-c-code-editor__code";

--- a/js/apps/admin-ui/cypress/support/pages/admin-ui/ListingPage.ts
+++ b/js/apps/admin-ui/cypress/support/pages/admin-ui/ListingPage.ts
@@ -72,8 +72,12 @@ export default class ListingPage extends CommonElements {
   }
 
   showNextPageTableItems() {
-    cy.get(this.#nextPageBtn).scrollIntoView();
-    cy.get(this.#nextPageBtn).click();
+    cy.get("body").then(($body) => {
+      if (!$body.find('[data-testid="' + this.#nextPageBtn + '"]').length) {
+        cy.get(this.#nextPageBtn).scrollIntoView();
+        cy.get(this.#nextPageBtn).click();
+      }
+    });
 
     return this;
   }

--- a/js/apps/admin-ui/cypress/support/pages/admin-ui/SidebarPage.ts
+++ b/js/apps/admin-ui/cypress/support/pages/admin-ui/SidebarPage.ts
@@ -17,13 +17,29 @@ export default class SidebarPage extends CommonElements {
   #identityProvidersBtn = "#nav-item-identity-providers";
   #userFederationBtn = "#nav-item-user-federation";
 
+  realmsElements = '[data-testid="realmSelector"] li';
+
   showCurrentRealms(length: number) {
     cy.findByTestId(this.#realmsDrpDwn).click();
-    cy.get('[data-testid="realmSelector"] li').should(
+    cy.get(this.realmsElements).contains("Loading realms…").should("not.exist");
+    cy.get(this.realmsElements).should(
       "have.length",
       length + 1, // account for button
     );
     cy.findByTestId(this.#realmsDrpDwn).click({ force: true });
+
+    return this;
+  }
+
+  realmExists(realmName: string, exists = true) {
+    cy.findByTestId(this.#realmsDrpDwn).click();
+    cy.get(this.realmsElements).contains("Loading realms…").should("not.exist");
+    cy.get(this.realmsElements)
+      .contains(realmName)
+      .should((exists ? "" : "not.") + "exist");
+    cy.findByTestId(this.#realmsDrpDwn).click();
+
+    return this;
   }
 
   getCurrentRealm() {
@@ -32,11 +48,9 @@ export default class SidebarPage extends CommonElements {
 
   goToRealm(realmName: string) {
     this.waitForPageLoad();
-    cy.findByTestId(this.#realmsDrpDwn)
-      .click()
-      .parent()
-      .contains(realmName)
-      .click();
+    cy.findByTestId(this.#realmsDrpDwn).click();
+    cy.get(this.realmsElements).contains("Loading realms…").should("not.exist");
+    cy.get(this.realmsElements).contains(realmName).click();
     this.waitForPageLoad();
 
     return this;

